src/AudacityException.cpp

```diff
@@ -73,17 +73,20 @@ void MessageBoxException::DelayedHandlerAction()
       // give the user no useful added information.
       
       if ( wxAtomicDec( sOutstandingMessages ) == 0 ) {
+         // If we have sentry reporting enabled - we should always show the
+         // ErrorDialog
+#ifndef HAS_SENTRY_REPORTING
          if (ErrorHelpUrl().IsEmpty())
          {
             ::AudacityMessageBox(
                ErrorMessage(),
                (caption.empty() ? AudacityMessageBoxCaptionStr() : caption),
-               wxICON_ERROR 
-            );
+               wxICON_ERROR);
          }
          else
+#endif
          {
-            ShowErrorDialog(
+            ShowExceptionDialog(
```

Comment:
This will cause the exception dialog whenever there is any AudacityException.

I think we may not want it for every exception, and might need to construct exception objects with an option not to use this dialog.

For instance exceptions in OnPaste in EditMenus.cpp.  You can get them if you select two wave tracks, copy, then select a wave and a label track, and try to paste.  Nothing goes wrong with external devices.  Rather the exception provides a way to write the complicated paste operation and allow backing out of some deeply nested place when the operation can't complete just for logical reasons that are discovered late.  It relies on a catch handler in AudacityApp::OnExceptionInMainLoop and the undo manager to restore document state.

Reply:
Other examples of SimpleMessageBoxException in WaveTrack.cpp.